controllers/platformoperator_controller.go

```diff
@@ -125,47 +149,57 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	return ctrl.Result{}, nil
 }
 
-func (r *PlatformOperatorReconciler) ensureBundle(ctx context.Context, po *platformopenshiftiov1alpha1.PlatformOperator, bundleName, bundleImage string) error {
-	if po.Status.InstalledBundleName != "" {
-		return nil
-	}
-	bundle := &rukpakv1alpha1.Bundle{}
-	if err := r.Get(ctx, types.NamespacedName{Name: bundleName}, bundle); err != nil {
+func (r *PlatformOperatorReconciler) ensureBundleInstance(ctx context.Context, po *platformopenshiftiov1alpha1.PlatformOperator, bundle *api.Bundle) error {
+	bundleName, bundleImage := bundle.CsvName, bundle.BundlePath
+	bi := &rukpakv1alpha1.BundleInstance{}
+	if err := r.Get(ctx, types.NamespacedName{Name: bundleName}, bi); err != nil {
 		if !apierrors.IsNotFound(err) {
 			return err
 		}
-		bundle = &rukpakv1alpha1.Bundle{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: bundleName,
-			},
-			Spec: rukpakv1alpha1.BundleSpec{
-				ProvisionerClassName: "core.rukpak.io/plain",
-				Source: rukpakv1alpha1.BundleSource{
-					Type: rukpakv1alpha1.SourceTypeImage,
-					Image: &rukpakv1alpha1.ImageSource{
-						Ref: bundleImage,
-					},
-				},
-			},
+		if err := controllerutil.SetOwnerReference(po, bi, r.Scheme); err != nil {
+			return err
 		}
-		if err := controllerutil.SetOwnerReference(po, bundle, r.Scheme); err != nil {
+		if err := r.createBundleInstance(ctx, bundleName, bundleImage); err != nil {
 			return err
 		}
-		if err := r.Create(ctx, bundle); err != nil {
+		po.Status.InstalledBundleInstanceName = bi.Name
+	} else if po.Status.InstalledBundleInstanceName != bundleName {
+		// TODO: Does this guarantee that this is a newer Bundle?
+		// Update the BundleInstance if a higher semver was found
+		if err := r.updateBundleInstance(ctx, bi, bundleName, bundleImage); err != nil {
 			return err
 		}
 	}
-	po.Status.InstalledBundleName = bundleName
 
 	return nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *PlatformOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&platformopenshiftiov1alpha1.PlatformOperator{}).
-		Watches(&source.Kind{Type: &operatorsv1alpha1.CatalogSource{}}, handler.EnqueueRequestsFromMapFunc(requeuePlatformOperators(mgr.GetClient()))).
-		Complete(r)
+// updateBundleInstance is responsible for taking a BundleInstance and pivoting to a new Bundle
+func (r *PlatformOperatorReconciler) updateBundleInstance(ctx context.Context, bi *rukpakv1alpha1.BundleInstance, name, image string) error {
```

Comment:
I'd inline this logic into the `r.ensureBundleInstance(...)` method but this is fine for now.